src/objects.rs

```diff
@@ -71,14 +71,67 @@ impl Id<Network> for Network {
 }
 impl_codes!(Network);
 
+#[derive(Clone, Debug)]
+pub struct Rgb {
+    pub red: u8,
+    pub green: u8,
+    pub blue: u8,
+}
+
+impl ::serde::Serialize for Rgb {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: ::serde::Serializer,
+    {
+        let color = format!("{:02X}{:02X}{:02X}", self.red, self.green, self.blue);
+        serializer.serialize_str(&color)
+    }
+}
+
+impl<'de> ::serde::Deserialize<'de> for Rgb {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: ::serde::Deserializer<'de>,
+    {
+        use serde::de::Error;
+
+        let color_hex = String::deserialize(deserializer)?;
+        let color_dec = u32::from_str_radix(&color_hex, 16).map_err(Error::custom)?;
+
+        if color_dec >= 1 << 24 {
+            return Err(Error::custom(format!(
+                "color should be in hexadecimal format (ex: FF4500)"
+            )));
+        }
+
+        Ok(Rgb {
+            red: (color_dec >> 16) as u8,
+            green: (color_dec >> 8) as u8,
+            blue: color_dec as u8,
+        })
+    }
+}
+
 #[derive(Serialize, Deserialize, Debug)]
 pub struct Line {
     #[serde(rename = "line_id")] pub id: String,
-    #[serde(rename = "line_name")] pub name: String,
+    #[serde(rename = "line_code")] pub code: Option<String>,
     #[serde(skip)] pub codes: CodesT,
+    #[serde(rename = "line_name")] pub name: String,
+    #[serde(rename = "forward_line_name")] pub forward_name: Option<String>,
+    pub forward_direction: Option<String>,
+    #[serde(rename = "backward_line_name")] pub backward_name: Option<String>,
+    #[serde(rename = "line_color")] pub color: Option<Rgb>,
+    #[serde(rename = "line_text_color")] pub text_color: Option<Rgb>,
+    #[serde(rename = "line_sort_order")] pub sort_order: Option<u32>,
     pub network_id: String,
     pub commercial_mode_id: String,
+    pub geometry_id: Option<String>,
+    //TODO Type Datetime?
```

Comment:
Yeah, but we'll do that later when reading stop times. Something using a u32 for the number of seconds. It'll need to be compact for memory considerations.